cmd/http/close.go

```diff
@@ -44,9 +44,7 @@ func DrainBody(respBody io.ReadCloser) {
 		// Without this closing connection would disallow re-using
 		// the same connection for future uses.
 		//  - http://stackoverflow.com/a/17961593/4465767
-		bufp := b512pool.Get().(*[]byte)
```

Comment:
This implementation was because we frequently drain the body due to the extensive usage of the HTTP client. So this code solves two things.

- One is to avoid using internal buffers used by io.Copy() repeatedly.
- Another is to use a smaller buffer 512 is enough to drain the buffer because practically we would have read the entire body anyways. But we need to do this to satisfy Go connection pooling requirements. So we don't need a 4k buffer we can simply use a 512 buffer.

Changing this to io.Copy is simpler but it adds lots of internal allocations..

Reply:
This is still inefficient. It doesn't make sense to use a fixed size buffer from a pool, when most of the time this call is preceded by a `ioutil.ReadAll()`. We are going through a atomic acquisition, and potentially adding the cost of allocating a 512 byte buffer for each close. 

Add to that, the close is called on an empty stream in most cases - and yet acquiring a buffer. 

The detail most people on stackoverflow and "internet" miss is that io.Copy is just fine because resp.Body is a *io.LimitedReader, and Copy does not allocate a 32K buffer in this case. In the most likely scenario of this being called on an already closed stream, io.Copy will not even allocate the 512 bytes.

The cost of acquiring a buffer on a previously read stream, along with the cost of allocating a new buffer even when it is not needed is a performance penalty, not an improvement.

Reply:
As discussed since io.Copy() does magic inside when its an io.LimitReader() we can use io.Copy() here and it will be simpler.